In [2]:
import pandas as pd
from sqlalchemy import create_engine
from scipy import stats

In [3]:
engine = create_engine(
"postgresql://postgres:root@localhost:5432/ecommerce_analytics"
)

In [4]:
payments = pd.read_sql("SELECT * FROM fact_payments;", engine)
order_items = pd.read_sql("SELECT * FROM fact_order_items;", engine)

In [5]:
order_value = order_items.groupby("order_id")["price"].sum().reset_index()

In [6]:
data = payments.merge(order_value, on="order_id")

In [7]:
credit_card = data[data["payment_type"] == "credit_card"]["price"]
boleto = data[data["payment_type"] == "boleto"]["price"]

In [8]:
t_stat, p_value = stats.ttest_ind(credit_card, boleto)

print(p_value)

5.63494190960879e-38


In [11]:
result = pd.DataFrame({
    "payment_type": ["credit_card", "boleto"],
    "avg_order_value": [credit_card.mean(), boleto.mean()],
    "orders": [len(credit_card), len(boleto)],
    "p_value": [p_value, p_value]
})

result

,payment_type,avg_order_value,orders,p_value
0,credit_card,143.873165,76278,5.634942e-38
1,boleto,121.929523,19614,5.634942e-38


In [12]:
print("Credit card orders:", len(credit_card))
print("Boleto orders:", len(boleto))

Credit card orders: 76278
Boleto orders: 19614


In [13]:
result.to_csv("../data_outputs/payment_method_test.csv", index=False)

Hypothesis Test Result

H0: Average order value is the same for credit_card and boleto payments.
H1: Average order value differs between payment methods.

The average order value for credit card payments is 143.87, while boleto payments average 121.93.

The p-value is 5.63e-38, which is far below the 0.05 significance threshold.

Therefore, we reject the null hypothesis and conclude that payment method is associated with different spending behavior. Customers paying with credit cards spend significantly more per order than boleto users.